### PR TITLE
adding stale bot config file

### PR DESCRIPTION
**Pull request by esc**
As documented in https://github.com/apps/stale.

### CODE REVIEW

**Comment by esc** [comment]
The exclusion labels and maybe the 'stale' label need a review. 'wontfix' does exist as a label in the Numba repository, but I think it has gone out of use somewhat.

File: .github/stale.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - bug - failure to compile
+  - bug - ice
+  - bug - incorrect behaviour
+  - bug - memory leak
+  - bug - miscompile
+  - bug - regression
+  - bug - segfault
+  - feature_request
+# Label to use when marking an issue as stale
+staleLabel: wontfix
```
Comment (stuartarchibald) [inline .github/stale.yml:17]: Agree with comments in the PR proper, think a `stale` label might be more appropriate? It's not that we `won't fix`, more that just something is probably lacking input/information to permit a classification of the problem?

File: .github/stale.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# Comment to post when marking an issue as stale. Set to `false` to disable
+markComment: >
+  This issue has been automatically marked as stale because it has not had
+  recent activity. It will be closed in 7 days if no further activity occurs.
+# Comment to post when closing a stale issue. Set to `false` to disable
+closeComment: >
+  This issue has been automatically closed as it was stale
```
Comment (stuartarchibald) [inline .github/stale.yml:24]:
```suggestion
  This issue has been automatically closed as it was stale.
```

File: .github/stale.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+exemptLabels:
+  - bug
+  - bug - failure to compile
+  - bug - ice
+  - bug - incorrect behaviour
+  - bug - memory leak
+  - bug - miscompile
+  - bug - regression
+  - bug - segfault
+  - feature_request
```
Comment (stuartarchibald) [inline .github/stale.yml:15]: should probably add the `performance*`, `enhancement` and `task` labels too? I think the `stale`ness will most likely apply to issue which are not successful in triage due to lack of information.

Any thoughts on the `question` label? Am wondering if this bot could close questions after e.g. a week of inactivity and also mention that `discourse` as the place to keep the Q&A corpus?

File: .github/stale.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Number of days of inactivity before an issue becomes stale
+daysUntilStale: 30
```
Comment (stuartarchibald) [inline .github/stale.yml:2]: Perhaps make this 2 weeks and see how that goes? I've not often seen issues that are not yet triaged and also being resolved after a couple of weeks, perhaps the bot commenting earlier about the staleness might prompt a response on e.g. missing information?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for adding this patch to help with the issue tracker. Suggestions/questions are inline.

File: .github/stale.yml (hunk 1 above)
Comment (esc) [reply .github/stale.yml:17]: right, so let us make a new label `stale` and then use that. I am not sure this label will see much use anyway?

Comment (stuartarchibald) [reply .github/stale.yml:17]: Done! https://github.com/numba/numba/labels/stale

Comment (esc) [reply .github/stale.yml:17]: Oh, we have a `stale` label?  Does anyone know if it has been there all along or did someone make it recently?

Comment (esc) [reply .github/stale.yml:17]: It looks like the `stale` label is not used in any issues, so let's use that.

Comment (stuartarchibald) [reply .github/stale.yml:17]: Made it for the purpose of this PR!

File: .github/stale.yml (hunk 3 above)
Comment (esc) [reply .github/stale.yml:15]: I'll add the labels yes.

But I don't think this bot supports these custom kinds of messages yet.

**Comment by esc** [comment]
@stuartarchibald I have addressed the comments, maybe @sklam or @seibert would like to take another look.

File: .github/stale.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  - bug - memory leak
+  - bug - miscompile
+  - bug - regression
+  - bug - segfault
+  - performance - compile time
+  - performance - run time
+  - performance
+  - feature_request
+  - enhancement
+  - task
```
Comment (stuartarchibald) [inline .github/stale.yml:20]: Am wondering also if this needs to be focused on issues only, else PRs that take a while get closed? e.g. https://github.com/probot/stale/blame/master/README.md#L68-L78 ?

Comment (esc) [reply .github/stale.yml:20]: yes, issues only,I'll go  and  fix it up.

Comment (esc) [reply .github/stale.yml:20]: added and pushed

**Comment by sklam** [comment]
I'm wondering if we should use `onlyLabels` or we need to do an label cleanup before enabling the bot. There are labels like "Numba1.0" that should not be stale.

**Comment by stuartarchibald** [comment]
@esc Should we close this for now?

**Comment by esc** [comment]
@stuartarchibald we might as well. I think StaleBot was a nice try, we learnt that the GitHub search and filter bar has bugs, we also learnt that Github never got back to us about about the filter and search being buggy. The positive outcome is that we manually pruned, cleaned and labeled our backlog once, which is good.

**Pull request closed without merging**